src/export_data.py

```diff
@@ -0,0 +1,145 @@
+"""
+
+# Can be run as follows:
+# python3 export_data.py --params_path=candid.json --export_folder=samples
+
+import json
+import argparse
+import ee
+from utils import TaskManager, SampleExporter, TropomiImagery, DSMImagery
+from utils import RoadImagery, MultiSpectralImagery, WindImagery, stack_bands_from_imagery
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description='Export TFRecords from EE')
+    parser.add_argument('--params_path', type=str,
+                        help="json parameters file location")
+    parser.add_argument('--start_date', type=str,
+                        help="start of the date range to export in YYYY-MM-DD",
+                        default="2018-01-01")
+    parser.add_argument('--end_date', type=str,
+                        help="end of the date range to export in YYYY-MM-DD",
+                        default="2019-12-31")
+    parser.add_argument('--params_path', type=str,
+                        help="json parameters file location")
+    parser.add_argument('--export_folder', type=str,
+                        help="Cloud export folder")
+    parser.add_argument('--bucket_name', type=str,
+                        help="cloud bucket name",
+                        default='aghriss-air-quality')
+    parser.add_argument('--num_samples', type=int,
+                        help="number of samples to export per image",
+                        default=1000)
+    parser.add_argument('--num_shards', type=int,
+                        help="number of sharding of samples before export",
+                        default=10)
+
+    args = parser.parse_args()
+    params_path = args.params_path
+    start_date = args.start_date
+    end_date = args.end_date
+    bucket_name = args.bucket_name
+    export_folder = args.export_folder
+    num_samples = args.num_samples
+    num_shards = args.num_shards
+
+    # params formated as folllows:
+    # params = {"collections" : collections_dict,
+    #           "bands" : bands_dict,
+    #           "kernel_radius" : int,
+    #           "scale" : int}
+    # both collections_dict and bands_dict have keys:
+    # ["multispectral","tropomi", "dsm", "wind", "road"]
+    # collection_dict values are collections' links
+    # bands_dict values are lists of bands to select from each collectionn
+
+    print("Loading", (params_path))
+    params = json.load(open(params_path, 'r'))
+    # patch_bands are the bands that will be exported as patches
+    patch_bands = []
+    patch_bands += params['bands']['multispectral']
+    patch_bands += params['bands']['tropomi']
+    patch_bands += params['bands']['road'] + params['bands']['dsm']
+    patch_bands += ["%i_%s" % (i, b) for b in params['bands']['wind']
+                    for i in range(12)]
+
+    all_bands = patch_bands + ["HOD", "DOW", "DOM", "MOY", "latitude",
+                               "longitude", "valid"]
+    scale = params['scale']
+    kernel_radius = params['kernel_radius']
```

Comment:
Config file parsing should be in its own method, also there are a lot of magic strings and a magic number here. Can we derive everything from the config file instead of having the parameter definitions be defined in two places (code and config file?)